lib/resolvers/author.ts

```diff
@@ -0,0 +1,153 @@
+import { Op, WhereOptions } from "sequelize";
+import { Author } from "../models/author";
+import { Book } from "../models/book";
+import { createAuthorLoader } from "../dataloaders/authorLoader";
+
+export const authorLoader = createAuthorLoader();
+
+export async function authorQueryResolver(
+  _parent: unknown,
+  {
+    limit = 10,
+    offset = 0,
+    id,
+    name,
+    born_year,
+  }: {
+    limit?: number;
+    offset?: number;
+    id?: string;
+    name?: string;
+    born_year?: string;
+  }
+) {
+  // If querying by ID, use the loader
+  if (id) {
+    const author = await authorLoader.load(id);
+    return {
+      authors: author ? [author] : [],
+      count: author ? 1 : 0,
+    };
+  }
+
+  const where: WhereOptions = {};
+
+  if (name)
+    where.name = {
+      [Op.iLike]: `%${name}%`,
+    };
+
+  if (born_year)
+    where.born_date = {
+      [Op.between]: [`${born_year}-01-01`, `${born_year}-12-31`],
+    };
+
+  const count = await Author.count({
+    where,
+  });
+
+  const authors = await Author.findAll({
+    where,
+    limit,
+    offset,
+    include: [{ model: Book, as: "books" }],
```

Comment:
There are different ways to avoid this, but its important to avoid. one way is function args, like we can have `includeBooks = false` as a arg, and do a join only if  required. 

But the better way is to use Data Loaders, with field resolvers. This will allow us to use graphql to its fullest, and give data only when required. 

So if you do 
```
query {
 getAuthors {
   books {
     name
    }
  }
}
```
Then only it would get books, otherwise it would not.